tokio-macros/src/lib.rs

```diff
@@ -37,22 +51,50 @@ pub fn main(_attr: TokenStream, item: TokenStream) -> TokenStream {
         return TokenStream::from(tokens);
     }
 
-    let result = quote! {
-        fn #name() #ret {
-            let mut rt = tokio::runtime::Runtime::new().unwrap();
-            rt.block_on(async { #body })
+    let mut runtime = RuntimeType::Multi;
+
+    for arg in args {
+        match arg {
+            syn::NestedMeta::Meta(syn::Meta::Word(ident)) => match ident.to_string().to_lowercase().as_str() {
+                "multi_thread" => runtime = RuntimeType::Multi,
+                "single_thread" => runtime = RuntimeType::Single,
+                name => panic!("Unknown attribute {} is specified", name),
+            },
+            _ => ()
+        }
+    }
+
+    let result = match runtime {
+        RuntimeType::Multi => quote! {
+            #(#attrs)*
+            fn #name() #ret {
+                let mut rt = tokio::runtime::Runtime::new().unwrap();
+                rt.block_on_async(async { #body })
```

Comment:
I think this was renamed `block_on`?

Reply:
Yep, didn't notice it to be changed already